CMakeLists.txt

```diff
@@ -41,3 +45,10 @@ if (ENABLE_TESTS)
   enable_testing()
   add_subdirectory(tests)
 endif()
+
+# Determine whether or not to generate documentation.
+OPTION(ENABLE_DOCS "Enable generation of doxygen-based documentation." OFF)
```

Comment:
Please use lower-case function names.
Upper-case is  (typically) for variables.
Reference: https://cliutils.gitlab.io/modern-cmake/chapters/intro/dodonot.html

Reply:
Done.